pkg/kapp/crdupgradesafety/change_validator.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2024 The Carvel Authors.
+// SPDX-License-Identifier: Apache-2.0
+package crdupgradesafety
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/openshift/crd-schema-checker/pkg/manifestcomparators"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ChangeValidation is a function that accepts a FieldDiff
+// as a parameter and should return:
+// - an error if the change would be unsafe
+// - a boolean representation of whether or not the change
+// has been fully handled (i.e no additional changes exist)
+type ChangeValidation func(diff FieldDiff) (bool, error)
+
+// EnumChangeValidation ensures that:
+// - No enums are added to a field that did not previously have
+// enum restrictions
+// - No enums are removed from a field
+// If either of the above scenarios are not satisfied an error
+// representing the failure will be returned.
+// Additionally, a boolean representation of whether or not the change
```

Comment:
Nit: It looks like the comment has been cut off abruptly.

Reply:
Good catch! I must've gotten distracted and forgot to finish it up 😅

Reply:
Should be fixed in https://github.com/carvel-dev/kapp/pull/921/commits/b192141d574799e7af39e98e4edd58dcf536cd93